src/main/java/frc/robot/subsystems/objectDetection/GamePieceTracker.java

```diff
@@ -0,0 +1,82 @@
+package frc.robot.subsystems.objectDetection;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.wpilibj.Timer;
+import java.util.LinkedList;
+import java.util.Optional;
+
+public class GamePieceTracker {
+
+  public static final double TIMESTAMP_THRESHOLD = 1.0;
+  public static final double NEW_TARGET_THRESHOLD = 0.33;
+
+  private static final LinkedList<Target> targets = new LinkedList<>();
+
+  public static void addTarget(double timestamp, Pose2d pose) {
+    Optional<Pose2d> currentFocusedGamePiece = getGamePiece();
+
+    if (currentFocusedGamePiece.isEmpty()) {
+      targets.add(new Target(timestamp, pose));
+      return;
+    }
+
+    Pose2d currentFocusedGamePiecePosition = currentFocusedGamePiece.get();
+    double distance =
+        currentFocusedGamePiecePosition.getTranslation().getDistance(pose.getTranslation());
+
+    if (distance < NEW_TARGET_THRESHOLD) {
+      targets.add(new Target(timestamp, pose));
+    } else {
+      targets.clear();
+      targets.add(new Target(timestamp, pose));
+    }
+  }
+
+  public static Optional<Pose2d> getGamePiece() {
```

Comment:
This gets the average location of the game piece over the last .33 seconds.  The name & documentation needs to make that clear.